mentoring/answer.py

```diff
@@ -24,39 +24,134 @@
 # Imports ###########################################################
 
 import logging
-
 from lazy import lazy
 
-from xblock.fragment import Fragment
-
-from .light_children import LightChild, Boolean, Scope, String
 from .models import Answer
-from .utils import render_template
+
+from xblock.core import XBlock
+from xblock.fields import Scope, Float, Integer, String
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+from .step import StepMixin
+import uuid
 
 
 # Globals ###########################################################
 
 log = logging.getLogger(__name__)
-
+loader = ResourceLoader(__name__)
 
 # Classes ###########################################################
 
-class AnswerBlock(LightChild):
+
+class AnswerMixin(object):
+    """
+    Mixin to give an XBlock the ability to read/write data to the Answers DB table.
+    """
+    def _get_course_id(self):
+        """ Get a course ID if available """
+        return getattr(self.runtime, 'course_id', 'all')
+
+    def _get_student_id(self):
+        """ Get student anonymous ID or normal ID """
+        try:
+            return self.runtime.anonymous_student_id
+        except AttributeError:
+            return self.scope_ids.user_id
+
+    def get_model_object(self, name=None):
+        """
+        Fetches the Answer model object for the answer named `name`
+        """
+        # By default, get the model object for the current answer's name
+        if not name:
+            name = self.name
+        # Consistency check - we should have a name by now
+        if not name:
+            raise ValueError('AnswerBlock.name field need to be set to a non-null/empty value')
+
+        student_id = self._get_student_id()
+        course_id = self._get_course_id()
+
+        answer_data, _ = Answer.objects.get_or_create(
+            student_id=student_id,
+            course_id=course_id,
+            name=name,
+        )
+        return answer_data
+
+    def validate_field_data(self, validation, data):
+        """
+        Validate this block's field data.
+        """
+        super(AnswerMixin, self).validate_field_data(validation, data)
+
+        def add_error(msg):
+            validation.add(ValidationMessage(ValidationMessage.ERROR, msg))
+
+        if not data.name:
+            add_error(u"A Question ID is required.")
+
+
+class AnswerBlock(AnswerMixin, StepMixin, StudioEditableXBlockMixin, XBlock):
```

Comment:
Hmm, I can't reproduce this, @Kelketek. I've tried answer blocks in old mongo and it works just fine. Perhaps the question ID your Answer block has is somehow invalid? Let me know if this problem persists, especially with a newly-created mentoring block and answer child. If it does persist, please check for errors in the JS/AJAX console.